File: peer-top20-report/api_client.py
"""从 ma.maoniux.com 获取已有报告数据"""

from __future__ import annotations

import json
from typing import Any

import requests

DEFAULT_API_BASE = "https://ma.maoniux.com/api/v1/report"


def fetch_report_by_id(report_id: int, *, api_base: str = DEFAULT_API_BASE) -> dict[str, Any]:
    response = requests.get(f"{api_base}/{report_id}", timeout=30)
    response.raise_for_status()
    payload = response.json()
    content = payload.get("content")
    if isinstance(content, str):
        return json.loads(content)
    if isinstance(content, dict):
        return content
    raise ValueError("报告详情中缺少 content 字段")


def load_json_file(path: str) -> dict[str, Any]:
    with open(path, "r", encoding="utf-8") as file:
        return json.load(file)

File: peer-top20-report/apiClient.js
import fs from 'node:fs';

const DEFAULT_API_BASE = 'https://ma.maoniux.com/api/v1/report';

export function loadJsonFile(path) {
  return JSON.parse(fs.readFileSync(path, 'utf-8'));
}

export async function fetchReportById(reportId, apiBase = DEFAULT_API_BASE) {
  const response = await fetch(`${apiBase}/${reportId}`);
  if (!response.ok) {
    throw new Error(`拉取报告失败: HTTP ${response.status}`);
  }
  const payload = await response.json();
  if (typeof payload.content === 'string') {
    return JSON.parse(payload.content);
  }
  if (typeof payload.content === 'object' && payload.content) {
    return payload.content;
  }
  throw new Error('报告详情中缺少 content 字段');
}
